### ISSUE TITLE
TensorFlow 2.6 installs Keras 2.7

### ISSUE THREAD

**Issue by guillaumekln** [tensorflow/tensorflow#52922]
**System information**
- OS Platform and Distribution: Ubuntu 18.04
- TensorFlow installed from: binary
- TensorFlow version: 2.6.1
- Python version: 3.6.9
- Installed using virtualenv? pip? conda?: virtualenv + pip

**Describe the problem**

TensorFlow 2.6 installs Keras 2.7, but it should install Keras 2.6 instead.

**Provide the exact sequence of commands / steps that you executed before running into the problem**

```bash
pip install tensorflow==2.6.*
```

**Any other info / logs**

See `keras-2.7.0` in the list of installed packages:

> `Successfully installed absl-py-0.15.0 astunparse-1.6.3 cached-property-1.5.2 cachetools-4.2.4 certifi-2021.10.8 charset-normalizer-2.0.7 clang-5.0 dataclasses-0.8 flatbuffers-1.12 gast-0.4.0 google-auth-2.3.3 google-auth-oauthlib-0.4.6 google-pasta-0.2.0 grpcio-1.41.1 h5py-3.1.0 idna-3.3 importlib-metadata-4.8.1 keras-2.7.0 keras-preprocessing-1.1.2 markdown-3.3.4 numpy-1.19.5 oauthlib-3.1.1 opt-einsum-3.3.0 protobuf-3.19.1 pyasn1-0.4.8 pyasn1-modules-0.2.8 requests-2.26.0 requests-oauthlib-1.3.0 rsa-4.7.2 six-1.15.0 tensorboard-2.7.0 tensorboard-data-server-0.6.1 tensorboard-plugin-wit-1.8.0 tensorflow-2.6.1 tensorflow-estimator-2.6.0 termcolor-1.1.0 typing-extensions-3.7.4.3 urllib3-1.26.7 werkzeug-2.0.2 wrapt-1.12.1 zipp-3.6.0`

**Comment by bhayden53** [comment]
Don't know if it's related but I'm seeing things like this in my own pip installation logs:

```
#9 70.10 Collecting tensorflow-estimator~=2.6
#9 70.15   Downloading tensorflow_estimator-2.7.0-py2.py3-none-any.whl (463 kB)
#9 74.69 Collecting tensorboard~=2.6
#9 74.72   Downloading tensorboard-2.7.0-py3-none-any.whl (5.8 MB)
```

I've noticed that `keras~=2.6` is also resolving to 2.7

Are these other dependencies being install incorrectly as well?

**Comment by sachinprasadhs** [comment]
The above linked PR is merged, now you should be able to see the compatible versions, please check [this](https://colab.research.google.com/gist/sachinprasadhs/3e16aa2c6b60c12bb3902957e9175b9d/52922.ipynb) gist for reference which is working as expected.Thanks.

**Comment by bhack** [comment]
@sachinprasadhs Your notebook is wrong as a Colab instance have a preinstalled env with Tensorflow and Keras that was prepared before Keras TF 2.7 was available so you cannot test in this way.

You need something like:
```
!pip uninstall -y tensorflow keras
!pip install tensorflow==2.6
```
or on a local instance:

```
docker run --rm -it python:3.9 pip install tensorflow==2.6
```

**Comment by bhack** [comment]
This is also against the official claimed Keras-TF versions compatibilities at:

https://github.com/keras-team/keras#release-and-compatibility

**Comment by mihaimaruseac** [comment]
This would be fixed in ~12 hours by a release of TF 2.6.2 patch release and TF 2.7.0 release.

The issue is a bug in TF 2.6 where we specified Keras dependency as `~= 2.6` instead of `~= 2.6.0`. The `~=` semantic is "take the version on the right, keep all the numbers specified there except the last one, that's the only one that can chance". So `~= 2.6` means that `keras == 2.99` would also be viable for the dependency resolution, clearly a bug.

This is not an issue with the stated Keras-TF compatibility. Just the `~=` semantic being buggy as it needs to specify `major.minor.patch` for these versions that are always in sync.

**Comment by bhack** [comment]
Yes exaclty as they claim:

> Keras 2.6 with TF >= 2.6 < 2.7

**Comment by mihaimaruseac** [comment]
This should be fixed with the release of TF 2.6.2
